src/ccbr_tools/jobby.py

```diff
@@ -1,804 +1,272 @@
+    exit_split = df["ExitCode"].str.split(":", expand=True)
+    df["ExitCode"] = pd.to_numeric(exit_split[0], errors="coerce").astype("Int64")
+    df["KillSignal"] = pd.to_numeric(exit_split[1], errors="coerce").astype("Int64")
 
-    # Job scheduler to query,
-    # available: SLURM, ...
-    # More coming soon!
-    parser.add_argument(
-        "-s",
-        "--scheduler",
-        type=lambda s: str(s).lower(),
-        required=False,
-        default="slurm",
-        choices=["slurm"],
-        help=argparse.SUPPRESS,
-    )
+    # Parse time columns to seconds
+    df["ElapsedSec"] = df["RunTime"].apply(parse_time_to_seconds)
+    df["CPUTimeSec"] = df["TotalCPUTime"].apply(parse_time_to_seconds)
 
-    # Define handlers for each sub-parser
-    parser.set_defaults(func=jobby)
-    # Parse command-line args
-    args = parser.parse_args()
+    # Ensure AllocCPUs is numeric
+    df["AllocCPUs"] = pd.to_numeric(df["NumCPUs"], errors="coerce")
 
-    return args
+    # Calculate CPU Efficiency
+    df["CPUEfficiency"] = np.where(
+        (df["ElapsedSec"] > 0) & (df["AllocCPUs"] > 0),
+        df["CPUTimeSec"] / (df["ElapsedSec"] * df["AllocCPUs"]),
+        np.nan,
+    )
 
+    # Optionally round CPUEfficiency to 2 decimals
+    df["CPUEfficiency"] = (df["CPUEfficiency"] * 100).round(
+        2
+    )  # Expressed as a percentage
+
+    # Reorder CPUEfficiency next to NumCPUs
+    cols = list(df.columns)
+    if "CPUEfficiency" in cols and "NumCPUs" in cols:
+        # Move CPUEfficiency right after NumCPUs
+        cols.insert(cols.index("NumCPUs") + 1, cols.pop(cols.index("CPUEfficiency")))
+        df = df[cols]
+
+    # Drop unwanted technical columns
+    df = df.drop(
+        columns=["TotalCPUTime", "AllocCPUs", "ElapsedSec", "CPUTimeSec"],
+        errors="ignore",
+    )
 
-def main():
-    # Sanity check for usage
-    if len(sys.argv) == 1:
-        # Nothing was provided
-        fatal("Invalid usage: {} [-h] [--version] ...".format(_name))
-
-    # Collect args for sub-command
-    args = parsed_arguments(name=_name, description=_description)
-
-    # Display version information
-    err("{} ({})".format(_name, __version__))
-    # Mediator method to call the
-    # default handler function
-    args.func(args)
+    # Reorder columns to keep KillSignal right after ExitCode
+    cols = list(df.columns)
+    if "KillSignal" in cols:
+        # Move KillSignal to immediately after ExitCode
+        cols.insert(cols.index("ExitCode") + 1, cols.pop(cols.index("KillSignal")))
+        df = df[cols]
+
+    # Output based on requested format
+    if output_format == "markdown":
+        print(df.to_markdown(index=False))
+    elif output_format == "tsv":
+        print(df.to_csv(sep="\t", index=False))
+    elif output_format == "json":
+        print(df.to_json(orient="records", indent=2))
+    elif output_format == "yaml":
+        print(yaml.dump(df.to_dict(orient="records"), sort_keys=False))
 
```

Comment:
suggestion: raise an error with a useful message if the output format is not supported
```suggestion
    # Output based on requested format
    if output_format == "markdown":
        print(df.to_markdown(index=False))
    elif output_format == "tsv":
        print(df.to_csv(sep="\t", index=False))
    elif output_format == "json":
        print(df.to_json(orient="records", indent=2))
    elif output_format == "yaml":
        print(yaml.dump(df.to_dict(orient="records"), sort_keys=False))
    else:
        raise ValueError(f"output format {output_format} not supported")
```